src/components/Card/Card.js

```diff
@@ -6,25 +9,66 @@ import "./__text/card__text.css";
 import "./__title/card__title.css";
 import "./__origin/card__origin.css";
 
-import image from "../../images/image_08.png";
+import "./__like-flag/card__like-flag.css";
+import "./__like-flag/_active/card__like-flag_active.css";
+import "./__delete-trash/card__delete-trash.css";
+import "./__category/card__category.css";
+import "./__button-description/card__button-description.css";
+import "./__button-description/_non-active/card__button-description_non-active.css";
+import MainApi from "../../utils/MainApi";
+
+function Card({
+  card,
+  isLoggedIn,
+  keyword,
+  updateSavedArticles,
+  savedArticles,
+}) {
+  const [isShown, setIsShown] = useState(false);
+  const [isLiked, setIsLiked] = useState(false);
+  const isAlreadySaved = savedArticles.some((c) => c.title === card.title);
 
-function Card({ children }) {
   return (
     <li className="card">
-      <img className="card__picture" src={image} alt={`A Card name`} />
+      <img className="card__picture" src={`${card.urlToImage}`} alt={""} />
       <div className="card__written-content-block">
-        <p className="card__date"> November 4, 2020 </p>
-        <h3 className="card__title">
-          Everyone Needs a Special 'Sit Spot' in Nature
-        </h3>
-        <p className="card__text">
-          Ever since I read Richard Louv's influential book, "Last Child in the
-          Woods," the idea of having a special "sit spot" has stuck with me.
-          This advice, which Louv attributes to nature educator Jon Young, is
-          for both adults and children to find...
+        <p className="card__date"> {moment(card.publishedAt).format("ll")} </p>
+        <h3 className="card__title"> {card.title} </h3>
+        <p className="card__text"> {card.description} </p>
+        <p className="card__origin"> {card.source.name} </p>
+        <button
+          className={`card__like-flag ${
+            isLiked || isAlreadySaved ? "card__like-flag_active" : ""
+          }`}
+          onClick={async (e) => {
+            if (isLoggedIn && !isLiked && !isAlreadySaved) {
+              try {
+                const newSavedArticle = await MainApi.saveArticle({
```

Comment:
please, make all api requests only in `App` and pass only `onSubmit` or `onClick` to the props